syfertext/language.py

```diff
@@ -101,12 +101,13 @@ def make_doc(self, text: Union[str, String, StringPointer]):
         are stored in a Doc object which is then returned.
         """
 
+        location_id = text.location.id
+
         # TODO: huge bugs in this function. deal with the case when the function a local string `text`
         #       after it has been already called with a remote string
-        if self.tokenizer is None:
-
+        if not location_id in self.tokenizers:
             # Create the Tokenizer object
-            self.tokenizer = self.factories["tokenizer"](
+            self.tokenizers[location_id] = self.factories["tokenizer"](
                 self.vocab,
                 owner=self.owner,
                 client_id=self.owner.id,  # This is the id of the owner of the Language object using this tokenizer
```

Comment:
Since at this point, we know the location on which the tokenizer should be sent, we can send it here (in case the text resides on a remote worker):
```
self.tokenizers[location_id] = self.tokenizers[location_id].send(<remote_worker>)
```